### DISCUSSION TITLE
Help on making niri black and white (grayscale) to improve concentration

### DISCUSSION THREAD

**Discussion by thepragandsensdiary** [YaLTeR/niri#911] [Q&A] [answered]
I have my Xiaomi MIUI phone with a grayscale filter and reading mode enabled, it looks amazing with it's AMOLED screen.
![VID-20241230-WA0006.webm](https://github.com/user-attachments/assets/21e7835a-ab19-44c3-8e84-cf1a2a002361)
Makes my phone less distracting and less appetizing to doomscroll.

I was wondering if I could replicate a similar look on wayland now that the new Lenovo Legion Go is about to come out in a few months, I've read on [reddit something about shaders on Hyprland](https://www.reddit.com/r/hyprland/comments/1es6xdk/how_to_have_greyscale_in_hyprland/), I recall having seen something about shaders on this [niri github wiki page](https://github.com/YaLTeR/niri/wiki/Configuration:-Animations). It could be done with the current version? `niri --version` `niri 0.1.10-1 (unknown commit)
`?

I'm using `Arch Linux` btw that's really important to know.

Best of wishes to Dev Yalter <3

**Comment by thepragandsensdiary** [comment]
Could the wayland protocol `wlr-gamma-control-unstable-v1` help to achieve my goal?? I'll link also the app [wlsunset](https://sr.ht/~kennylevinsen/wlsunset/). Pwease hewp TnT.

**Comment by YaLTeR** [comment]
Hey, currently there's no support for long-running shaders, but once they're implemented in niri, it should be fairly straightforward to do black and white. I don't think gamma control can work for this because it acts on color components independently.

**Reply by thepragandsensdiary** [reply]
tenkiu dev yalter, i'm a big fan. I'll be on the lookout~

**Reply by YaLTeR** [reply, marked as answer]
Opened an issue: https://github.com/YaLTeR/niri/issues/913

**Reply by thepragandsensdiary** [reply]
Also, unrelated but pretty cute~, while I was following this discussion I finished my first "split" (there are 12 enameled thin wires in the middle cable haha) Dactyl Manuform, I'll try to test some tapdance and Stenography shenanigans on wayland niri for the forseeable future, pretty proud of my build btw.
![IMG_20250102_015258](https://github.com/user-attachments/assets/121d7188-fbd7-4435-ab1b-bd700b0ce315)
I'll learn rust and contribute I promiss >n<

**Reply by YaLTeR** [reply]
Looks cool! Tell me how it goes. I personally can't stand most mod-taps myself due to the added latency, but I do use one on a button where it doesn't matter.
